src/MSALWrapper/AuthFlow/Broker.cs

```diff
@@ -84,57 +77,52 @@ protected override async Task<(TokenResult, IList<Exception>)> GetTokenInnerAsyn
-                try
+                tokenResult = await CachedAuth.GetTokenAsync(
+                    this.logger,
+                    this.scopes,
+                    account,
+                    this.pcaWrapper,
+                    this.errors);
+
+                if (tokenResult != null)
                 {
-                    try
-                    {
-                        var tokenResult = await TaskExecutor.CompleteWithin(
-                            this.logger,
-                            this.silentAuthTimeout,
-                            "Get Token Silent",
-                            (cancellationToken) => this.pcaWrapper.GetTokenSilentAsync(this.scopes, account, cancellationToken),
-                            this.errors)
-                            .ConfigureAwait(false);
-                        tokenResult.SetSilent();
-
-                        return (tokenResult, this.errors);
-                    }
-                    catch (MsalUiRequiredException ex)
-                    {
-                        this.errors.Add(ex);
-                        this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
-                        var tokenResult = await TaskExecutor.CompleteWithin(
-                            this.logger,
-                            this.interactiveAuthTimeout,
-                            "Interactive Auth",
-                            (cancellationToken) => this.pcaWrapper
-                            .WithPromptHint(this.promptHint)
-                            .GetTokenInteractiveAsync(this.scopes, account, cancellationToken),
-                            this.errors)
-                            .ConfigureAwait(false);
+                    return (tokenResult, this.errors);
+                }
+
+                Func<CancellationToken, Task<TokenResult>> getTokenInteractive = (cancellationToken) =>
+                    this.pcaWrapper
+                        .WithPromptHint(this.promptHint)
+                        .GetTokenInteractiveAsync(this.scopes, account, cancellationToken);
 
-                        return (tokenResult, this.errors);
-                    }
+                try
+                {
+                    tokenResult = await TaskExecutor.CompleteWithin(
+                        this.logger,
+                        this.interactiveAuthTimeout,
+                        $"{this.Name()} interactive auth",
+                        getTokenInteractive,
+                        this.errors).ConfigureAwait(false);
                 }
                 catch (MsalUiRequiredException ex)
                 {
                     this.errors.Add(ex);
-                    this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
-                    var tokenResult = await TaskExecutor.CompleteWithin(
+                    this.logger.LogDebug($"initial {this.Name()} auth failed. Trying again with claims from exception.\n{ex.Message}");

```

Comment:
We can assert these logs in the tests.

Reply:
We can, but we do not currently. I agree it would be good to make log assertions here and I'd like to add those after the impending test refactors since _most_ of roughly half of the tests will be moved or deleted.

Reply:
@shalinikhare27 does that work for you?

Reply:
I've opened #282 to track that.